packages/react-native-web/src/exports/TextInput/index.js

```diff
@@ -243,7 +243,8 @@ const TextInput: React.AbstractComponent<
       }
       if (selectTextOnFocus) {
         // Safari requires selection to occur in a setTimeout
-        setTimeout(() => {
+        clearTimeout(TextInput.focusTimeout);
+        TextInput.focusTimeout = setTimeout(() => {
```

Comment:
The timeout should be managed as a ref or callback instead of this approach

Reply:
If we used refs we'd only have access to this instance's timeout. But we need to clear a timeout created by another TextInput instance.